stan/math/prim/meta/require_generics.hpp

```diff
@@ -629,6 +644,26 @@ using require_all_not_eigen_t = require_all_not_t<is_eigen<Types>...>;
 template <typename... Types>
 using require_any_not_eigen_t = require_any_not_t<is_eigen<Types>...>;
 
+template <typename T>
```

Comment:
These are really neat.  

Was the decision that they dont need doc because they just follow the definitions and the standard library doesn't doc the `_t` types?  These are a bit different in that there's not underlying simple structs they'e wrapping.   I'd rather err in favor of at least simple doc for compounds like this.  Especially since the whole notion of "Eigen matrix" is fundamentally confused given that vectors and row vectors use a specialization of the `Matrix` type.

Reply:
Yeah in another PR I'm going to break these out into the respective meta functions so that the requires base stuff just sits in `require_generics.hpp` and the specializations are in each `is_*` file